metaflow/plugins/kfp/kfp.py

```diff
@@ -1,316 +1,419 @@
-
-    # Note:
-    # Level-order traversal is adopted to keep the task-ids in line with what happens during a local metaflow execution.
-    # It is not entirely necessary to keep this order of task-ids if we are able to point to the correct input-paths for
-    # each step. But, using this ordering does keep the organization of data in the datastore more understandable and
-    # natural (i.e., `start` gets a task id of 1, next step gets a task id of 2 and so on with 'end' step having the
-    # highest task id. So the paths in the datastore look like: {run-id}/start/1, {run-id}/next-step/2, and so on)
-    """
-
-    def build_cmd_template(step_name, task_id, input_paths):
+        self.name = name
+        self.graph = graph
+        self.flow = flow
+        self.code_package = code_package
+        self.code_package_url = code_package_url
+        self.metadata = metadata
+        self.datastore = datastore
+        self.environment = environment
+        self.event_logger = event_logger
+        self.monitor = monitor
+        self.namespace = namespace
+        self.username = username
+
+        self._client = kfp.Client(namespace=api_namespace, userid=username, **kwargs)
+
+    def create_run_on_kfp(self, experiment_name, run_name):
         """
-        Returns the python command template to be used for each step.
-
-        This method returns a string with placeholders for `datastore_root` and `run_id`
-        which get populated using the provided config and the kfp run ID respectively.
-        The rest of the command string is populated using the passed arguments which are known before the run starts.
+        Creates a new run on KFP using the `kfp.Client()`.
+        """
+        run_pipeline_result = self._client.create_run_from_pipeline_func(
+            pipeline_func=self.create_kfp_pipeline_from_flow_graph(),
+            arguments={},
+            experiment_name=experiment_name,
+            run_name=run_name,
+            namespace=self.namespace,
+        )
+        return run_pipeline_result
+
+    def create_kfp_pipeline_yaml(self, pipeline_file_path=DEFAULT_KFP_YAML_OUTPUT_PATH):
+        """
+        Creates a new KFP pipeline YAML using `kfp.compiler.Compiler()`.
+        Note: Intermediate pipeline YAML is saved at `pipeline_file_path`
+        """
+        kfp.compiler.Compiler().compile(
+            self.create_kfp_pipeline_from_flow_graph(), pipeline_file_path
+        )
+        return os.path.abspath(pipeline_file_path)
 
-        An example constructed command template (to run a step named `hello`):
-        "python downloaded_flow.py --datastore s3 --datastore-root {ds_root} " \
-                         "step hello --run-id {run_id} --task-id 2 " \
-                         "--input-paths {run_id}/start/1"
+    @staticmethod
+    def _command(code_package_url, environment, step_name, step_cli):
```

Comment:
Can we add a more descriptive name that `_command` here? Perhaps `init_and_start_task_command`.

Reply:
I kept the function name to be the same as the one copied from `batch.py`.

Reply:
Sounds good.